server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -790,7 +790,9 @@ private void maybeSyncGlobalCheckpoints() {
     }
 
     private void syncRetentionLeases() {
-        sync(IndexShard::syncRetentionLeases, "retention lease");
+        if (indexSettings.isSoftDeleteEnabled()) {
+            sync(IndexShard::syncRetentionLeases, "retention lease");
```

Comment:
This can be tested with an IT test that created an index without soft deletes enabled, sets the sync interval to 100ms, and then hooks into the mock network layer and asserts no sync is ever sent?

Reply:
Nice suggestion. Thanks Jason.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/39922/commits/2e99cab686cc0f52ed1d007d4feb958527ac6fd0.